client/src/components/DisplayTable.js

```diff
@@ -71,50 +64,49 @@ export class DisplayTable extends React.Component {
     } = this.state;
 
     const clean_search_string = (search_string) => _.chain(search_string).deburr().toLower().trim().value();
-    const sorted_filtered_data = _.chain(rows)
-      .filter(
-        ({search_values}) => _.chain(search_values)
-          .map( (search_value, column_key) => (
-            _.isEmpty(searches[column_key]) ||
+    const sorted_filtered_data = _.chain(data)
+      .filter( (row) => _.chain(row)
+        .map( (column_value, column_key) => (
+          _.isEmpty(searches[column_key]) ||
             _.includes(
-              clean_search_string(search_value),
+              clean_search_string(column_value),
               clean_search_string(searches[column_key])
             )
-          ) )
-          .every()
-          .value()
+        ) )
+        .every()
+        .value()
       )
-      .sortBy( ({sort_values}) => _.isNumber(sort_values[sort_by]) ? sort_values[sort_by] : Number.NEGATIVE_INFINITY )
+      .sortBy( row => ( row[sort_by] && _.isNumber(row[sort_by]) )
+        ? row[sort_by].value : Number.NEGATIVE_INFINITY )
       .tap( descending ? _.reverse : _.noop )
       .value();
-
-    const all_sort_keys = _.chain(rows)
-      .flatMap( row => _.keys(row.sort_values) )
-      .uniq()
-      .value();
-
-    const all_search_keys = _.chain(rows)
-      .flatMap( row => _.keys(row.search_values) )
-      .uniq()
-      .value();
     
     const total_row = _.reduce(
       sorted_filtered_data,
       (totals, row) => _.mapValues(
         totals,
-        (total, col_key) => total + row.sort_values[col_key]
+        (total, col_key) => total + row[col_key]
       ),
-      _.mapValues(total_row_config, () => 0)
+      _.chain(column_configs)
+        .pickBy(col => col.is_summable)
+        .mapValues(col => 0)
+        .value()
     );
+    const is_total_exist = !_.isEmpty(total_row);
+
+    const ordered_column_keys = _.reduce( column_configs, (ordered_arr, column, column_key) => {
+      ordered_arr.splice(column.index, 0, column_key);
```

Comment:
Trying to make ordered column keys here. Not sure if this is the right approach

Reply:
Definitely an interesting approach, works I think, but not one I would have thought of! Array will have blanks in it if an index is skipped by accident in the column config (I commented on a case like that further down, should check if that actually breaks anything in here).

I'd probably write it something like 
```
  _.chain(column_configs)
    .map( ({index}, key) => [index, key] )
    .sortBy( _.first )
    .map( _.last )
    .value()
```

Reply:
ohhh ha I never thought of sorting like this. But this is definitely something I wanted! Thank you